runtime/near-vm-runner/src/runner.rs

```diff
@@ -146,3 +149,77 @@ impl VMKindExt for VMKind {
         }
     }
 }
+
+pub struct FilesystemCompiledContractCache {
+    dir: rustix::fd::OwnedFd,
+}
+
+impl FilesystemCompiledContractCache {
+    pub fn new() -> Self {
+        let path = std::env::var_os("NEAR_CONTRACT_CACHE");
+        let path = path.map(PathBuf::from).unwrap_or_else(|| {
+            PathBuf::from(std::env::var_os("HOME").expect("$HOME"))
+                .join(".near")
+                .join("data")
+                .join("contracts")
+        });
+        std::fs::create_dir_all(&path).expect("create contract directory");
+        let dir = rustix::fs::open(path, rustix::fs::OFlags::DIRECTORY, rustix::fs::Mode::empty())
+            .expect("open contract dir");
+        Self { dir }
+    }
+}
+
+/// Cache for compiled contracts code in plain filesystem.
+impl CompiledContractCache for FilesystemCompiledContractCache {
+    fn put(&self, key: &CryptoHash, value: CompiledContract) -> std::io::Result<()> {
+        use rustix::fs::{Mode, OFlags};
+        let final_filename = key.to_string();
+        let filename = format!("{}.temp", key);
```

Comment:
nit: can we rename to something like `tmp_filename` please?  It will help with readability.